src/services/web3/web3.plugin.ts

```diff
@@ -67,15 +68,19 @@ type PluginState = {
 };
 
 async function isSanctionedAddress(address: string): Promise<boolean | null> {
-  try {
-    const response = await axios.post(SANCTIONS_ENDPOINT, [
-      {
-        address: address.toLowerCase(),
-      },
-    ]);
-    const isSanctioned = response.data[0].isSanctioned;
-    return isSanctioned;
-  } catch {
+  if (configService.env.WALLET_SCREENING) {
+    try {
+      const response = await axios.post(SANCTIONS_ENDPOINT, [
+        {
+          address: address.toLowerCase(),
+        },
+      ]);
+      const isSanctioned = response.data[0].isSanctioned;
+      return isSanctioned;
+    } catch {
+      return null;
+    }
+  } else {
```

Comment:
Last question, unrelated to your work, why is the return type `Promise<boolean | null>` here? Could we not default to `false` if it fails? Is there any indication from the code why that decision was made?

Reply:
It's because it's an `async` function and therefore it returns a promise which must be resolved with `await`. It's an `async` function because it makes `axios.post` call.

Reply:
Yeah but that doesn't stop it returning false instead of null. It could be like this:
```ts
async function isSanctionedAddress(address: string): Promise<boolean> {
  if (configService.env.WALLET_SCREENING) {
    try {
      const response = await axios.post(SANCTIONS_ENDPOINT, [
        {
          address: address.toLowerCase(),
        },
      ]);
      const isSanctioned = response.data[0].isSanctioned;
      return isSanctioned;
    } catch {
      return false;
    }
  } else {
    return false;
  }
}
```

Reply:
I'm just wondering why it was returning null in the first place.

Reply:
Oh, sorry, I misunderstood. Perhaps the response `isSanctioned` is not always boolean and can be empty?

Reply:
Nonetheless, can we fix this function so that it always returns a boolean please.

Reply:
Updated 🙂